src/partialBindTo.ts

```diff
@@ -0,0 +1,74 @@
+import { purry } from './purry';
+
+type RemovePrefix<
+  P extends ReadonlyArray<unknown>,
+  T extends ReadonlyArray<unknown>,
+> = P extends readonly [...T, ...infer U]
+  ? U
```

Comment:
1. These should probably also use IterableContainer to cover all cases (I can't think of one, but just in case).
2. Are the letters here reversed? P is for "Prefix"? shouldn't it be the other way around?
```suggestion
type RemovePrefix<
  T extends IterableContainer,
  Prefix extends IterableContainer,
> = T extends readonly [...Prefix, ...infer Rest]
  ? Rest
```